plugin.video.ytchannels/plugin.video.ytchannels/default.py

```diff
@@ -0,0 +1,709 @@
+
+
+    if next_page!='1':
+        uri = build_url({'mode': 'open_playlists', 'id': '%s'%id, 'page' : '%s'%next_page ,'playlist':'yes'})
+        li = xbmcgui.ListItem('Next Page >>', iconImage=thumb)
+        xbmcplugin.addDirectoryItem(handle=addon_handle, url=uri, listitem=li,isFolder=True)
+    xbmcplugin.endOfDirectory(addon_handle)
+
+
+
+elif mode[0]=='add_channel':
+    options=['Search for channel','Add by username']
+    ind = xbmcgui.Dialog().select('Choose option', options)
+    if ind==0:
+
+        dicti=urlparse.parse_qs(sys.argv[2][1:])
+        foldername=dicti['foldername'][0]
+        
+        keyboard = xbmc.Keyboard('', 'Search channel:', False)
+        keyboard.doModal()
+        
+        if keyboard.isConfirmed():
+            channel_name = keyboard.getText()
+            results=search_channel(channel_name)
+            
+            result_list=[]
+            for i in range(len(results)):
+                result_list+=[results[i][0]]
+            dialog = xbmcgui.Dialog()
+            index = dialog.select('Choose channel', result_list)
+            if index>-1:
+                channel_uplid=results[index][1]
+                channel_name=results[index][0]
+                thumb=results[index][2]
+                channel_id=results[index][3]
+
+                add_channel(foldername,channel_name,channel_uplid,thumb)
+    elif ind==1:
+        dicti=urlparse.parse_qs(sys.argv[2][1:])
+        foldername=dicti['foldername'][0]
+        
+        keyboard = xbmc.Keyboard('', 'Enter username:', False)
+        keyboard.doModal()
+        
+        if keyboard.isConfirmed():
+            channel_name = keyboard.getText()
+            help_list=search_channel_by_username(channel_name)
+            
+            if help_list== 'not found':
+                xbmcgui.Dialog().ok("Youtube Channels", 'Channel with username "%s" does not exist.'%channel_name)
+            else:
+                channel_name=help_list[0]
+                channel_id=help_list[1]
+                thumb=help_list[2]
+                add_channel(foldername,channel_name,channel_id,thumb)
+
+    xbmc.executebuiltin("Container.Refresh")
+
+
+elif mode[0]=='rem_channel':
+    dicti=urlparse.parse_qs(sys.argv[2][1:])
+    channel_id=dicti['channel_id'][0]
+    remove_channel(channel_id)
+    xbmc.executebuiltin("Container.Refresh")
+
+elif mode[0]=='rem_folder':
+    dicti=urlparse.parse_qs(sys.argv[2][1:])
+    foldername=dicti['foldername'][0]
+    remove_folder(foldername)
+    xbmc.executebuiltin("Container.Refresh")
+
+elif mode[0]=='erase_all':
+    ret = xbmcgui.Dialog().yesno('Erase database', 'Do you wish to erase your Youtube Channels database?' )
+    if ret:
+        delete_database()
+        xbmcgui.Dialog().ok("Youtube Channels", "Successfully erased folder database!")
+
+
+elif mode[0]=='open_category':
+    dicti=urlparse.parse_qs(sys.argv[2][1:])
+    site=dicti['site'][0]
+
+    open_category(site)
+
+elif mode[0]=='import':
```

Comment:
Bad practice to keep all the plugin logic on the entry point file. This is not compiled to bytecode by Kodi python interpreter. Please.try to split up the code base and just import it in the entry point file